### DISCUSSION TITLE
Programmatically rebooting a node

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2357] [Q&A] [answered]
I'm probably missing something but I can't find it anywhere in the code or documentation.

1. What is the proper way to reboot a physical hardware mote?
2. Is it okay to do it to a Cooja node?

**Comment by rajeev1986** [comment]
Hi,
`SysCtrlSystemReset()` should be able to reset the hardware device for you in software.

**Comment by arurke** [comment, marked as answer]
Hi, a platform-agnostic approach would be `watchdog_reboot()`. However, it seems this is not implemented on Cooja.

**Reply by rahav** [reply]
Thank you @arurke what is the proper way to ask for this feature to be prioritized for Cooja?

**Reply by arurke** [reply]
I don't think there is any other way than entice/convince/hope someone in the community wants to have a look at it. With that said, I would not be surprised if a Cooja node can be rebooted from "outside", e.g. via the simulation script - perhaps @pjonsson is aware of any such options.

**Reply by pjonsson** [reply]
Having thought for 30 seconds, I can say it's not obvious to me how rebooting would be implemented in Cooja.

I have never used Cooja, but @nfi might be able to tell us if it's possible to trigger from the outside. My guess is that the node could be removed and a new one created in its place, but that will probably affect other things in the simulation as well.
